### DISCUSSION TITLE
Site working in dev mode but not on Cloudflare pages

### DISCUSSION THREAD

**Discussion by Maxson52** [sveltejs/kit#5375] [Q&A] [answered]
When deploying to Cloudflare pages with `adapter-auto` and navigating to the site, I see a 500 error displayed staying `Uncaught ReferenceError: global is not defined`.

I've encountered no issues in dev mode and when running `npm run preview`. Additionally, a test API route returns without issue.

Though I've never used Cloudflare pages before, I've deployed many sites with the exact same libraries and functionality to Netlify and Vercel without issue.

**Comment by Maxson52** [comment, marked as answer]
This error is generally found when using nodejs functionality in a non-node environment. In my case, it was using the dotenv framework.

**Comment by martinsotirov** [comment]
@Maxson52  So what's the solution?

**Reply by Maxson52** [reply]
Use Vercel or Netlify

**Reply by martinsotirov** [reply]
Well for me it turned out the solution is to remove the `'@nuxtjs/supabase'` module and now my Nuxt site works properly on Cloudflare Pages. I know it's not related to Svelte but it was the same error and I guess the same cause that you mentioned.
